### PR TITLE
feat: add autoclass feature for cloud storage

### LINKED ISSUE #220: Allow configure the autoclass argument
### TL;DR

Provision the "autoclass" argument in each bucket.

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket#autoclass
```

### Detailed design

```markdown
In the google_storage_bucket resource, we can specify whether enable "Autoclass" feature or not by the optional argument "autoclass".
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Ameausoone**
This PR provides the autoclass feature for simple_bucket and multiple_buckets. 

I didn't add a lot of things about tests, I didn't get yet how it works. 

Closes #220

### CODE REVIEW

**Comment by Ameausoone** [comment]
OK sorry for the first try, it couldn't work. 

OK now I fixed a lot of fixed, I couldn't run integration tests exactly like CI (I don't have access to an organisation where I could create projects). 

But I run integration tests : 
* I add a test with the module simple_bucket
* But I couldn't add a test with multiple_buckets because `autoclass` is exclusive with `lifecycle_rules` or I have to create a complete other tests.
* I hope lint is ok, I had to hack a little my project in local to run integration tests, so I couldn't run linter exactly like CI.

**Comment by Ameausoone** [comment]
Thks ! Is there a way to get access to logs ? My email is meausoone.a(at)sfeir.com ?

**Comment by Tensho** [comment]
Would be nice to have this 👍

**Comment by apeabody** [comment]
Pre-Req for CI: https://github.com/terraform-google-modules/terraform-google-cloud-storage/pull/257

**Comment by apeabody** [comment]
Hi @Ameausoone - You should now be able to view the lint results, however that isn't currently possible for the `terraform-google-cloud-storage-int-trigger ` tests.  Here is their most recent output:

```
Error: googleapi: Error 409: Your previous request to create the named bucket succeeded and you already own it., conflict

  with module.bucket.google_storage_bucket.bucket,
  on ../../modules/simple_bucket/main.tf line 17, in resource "google_storage_bucket" "bucket":
  17: resource "google_storage_bucket" "bucket" {
}
    apply.go:34: 
        	Error Trace:	/builder/home/go/pkg/mod/github.com/gruntwork-io/terratest@v0.43.3/modules/terraform/apply.go:34
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.0/pkg/tft/terraform.go:393
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.0/pkg/tft/terraform.go:403
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.0/pkg/tft/terraform.go:427
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.0/pkg/utils/stages.go:31
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.0/pkg/tft/terraform.go:427
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: googleapi: Error 409: Your previous request to create the named bucket succeeded and you already own it., conflict
        	            	
        	            	  with module.bucket.google_storage_bucket.bucket,
        	            	  on ../../modules/simple_bucket/main.tf line 17, in resource "google_storage_bucket" "bucket":
        	            	  17: resource "google_storage_bucket" "bucket" {
        	            	}
        	Test:       	TestAll/examples/simple_bucket
2023/07/05 18:30:02 RUN_STAGE env var set to apply
2023/07/05 18:30:02 Skipping stage teardown
--- FAIL: TestAll (19.41s)
    --- PASS: TestAll/examples/bucket_autoclass (14.58s)
    --- FAIL: TestAll/examples/simple_bucket (4.83s)
FAIL
FAIL	github.com/terraform-google-modules/terraform-google-cloud-storage/test/integration	19.436s
```

File: examples/bucket_autoclass/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| project\_id | The ID of the project in which to provision resources. | `string` | n/a | yes |
+
+## Outputs
+
+No output.
```
Comment (apeabody) [inline examples/bucket_autoclass/README.md:14]:
```suggestion
No outputs.
```

**Comment by fernandoiury** [comment]
any idea when this will be merged?

**Comment by fernandoiury** [comment]
@apeabody any idea when this is going to be merged? Thanks

**Comment by apeabody** [comment]
Hi @Ameausoone  @fernandoiury - The CI is currently raising this error

```
Error: googleapi: Error 409: Your previous request to create the named bucket succeeded and you already own it., conflict

  with module.bucket.google_storage_bucket.bucket,
  on ../../modules/simple_bucket/main.tf line 17, in resource "google_storage_bucket" "bucket":
  17: resource "google_storage_bucket" "bucket" {
}
    apply.go:34: 
        	Error Trace:	/builder/home/go/pkg/mod/github.com/gruntwork-io/terratest@v0.43.6/modules/terraform/apply.go:34
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.1/pkg/tft/terraform.go:393
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.1/pkg/tft/terraform.go:403
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.1/pkg/tft/terraform.go:427
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.1/pkg/utils/stages.go:31
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.6.1/pkg/tft/terraform.go:427
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: googleapi: Error 409: Your previous request to create the named bucket succeeded and you already own it., conflict
        	            	
        	            	  with module.bucket.google_storage_bucket.bucket,
        	            	  on ../../modules/simple_bucket/main.tf line 17, in resource "google_storage_bucket" "bucket":
        	            	  17: resource "google_storage_bucket" "bucket" {
        	            	}
        	Test:       	TestAll/examples/simple_bucket
```

**Comment by fernandoiury** [comment]
@apeabody no worries. I actually submitted a simplest PR to add this same feature. Could you take a look? Thanks https://github.com/terraform-google-modules/terraform-google-cloud-storage/pull/265

**Comment by Ameausoone** [comment]
thks @fernandoiury !

**Pull request closed without merging**